src/services/userLoginService.ts

```diff
@@ -5,54 +5,102 @@ import {
+  const [disable, setDisable] = useState(false);
+  const [loading, setLoading] = useState(false);
   const [formState, setFormState] = useState({
     email: "",
     password: "",
     errors: "",
   });
-    
-  const sendLogin = async (data: UserLoginDTO) => {
 
-    await validateForm(data)
-    try {
-      const response = await axios.post(
-        "https://mentores-backend.onrender.com/auth/login",
-        data
+  const handleError = (message: string) => {
+    toast.error(message, {
+      position: toast.POSITION.TOP_CENTER,
+      toastId: "customId",
+    });
+  };
+
+  const handleErrorNotifications = () => {
+    if (countError === 3) {
+      handleError(
+        'Você digitou a senha incorretamente e será bloqueado após cinco tentativas. Para cadastrar uma nova senha clique em "Esqueci minha senha".'
       );
-      setFormState({
-        ...formState,
-        errors: ""
-      })
-      router.push('/genericPage')
-    } catch (error) {
-      setFormState({
-        ...formState,
-        errors:"*E-mail ou senha incorretos."
-      })
-      setCountError(countError + 1)
-      console.log(countError)
+    }
+    if (countError >= 4) {
+      setSubmitButton(true);
+      handleError(
+        'Por questões de segurança, bloqueamos sua conta após você ter atingido a quantidade máxima de tentativas de acesso. Para cadastrar uma nova senha, clique em "Esqueci minha senha".'
+      );
+      setDisable(true);
+      setSubmitButton(true);
     }
   };
 
-  const validateForm = async(data:UserLoginDTO):Promise<boolean> => { 
-    if(!data.email || !data.password){
+  const sendLogin = async (data: UserLoginDTO) => {
+    const isValid = await validateForm(data);
+    if (isValid) {
+      setLoading(true);
+      try {
+        const response = await axios.post(
+          "https://mentores-backend.onrender.com/auth/login",
+          data
+        );
+        setFormState({
+          ...formState,
+          errors: "",
+        });
+        router.push("/genericPage");
+      } catch (error) {
+        setFormState({
+          ...formState,
+          errors: "*E-mail ou senha incorretos.",
+        });
+        setCountError(countError + 1);
+        handleErrorNotifications();
+      }
+    }
+    setTimeout(() => {
+      setLoading(false);
+    }, 500);
+  };
+
+  const validateForm = async (data: UserLoginDTO): Promise<boolean> => {
+    if (!data.email || !data.password) {
       setFormState({
         ...formState,
-        errors:"*E-mail ou senha incorretos."
-      })
-      return false
-    }else {
-      return true
-    }   
+        errors: "*E-mail ou senha incorretos.",
+      });
+      return false;
+    } else {
+      return true;
+    }
+  };
+
+  function checkFields(data: UserLoginDTO) {
+    return data.email.trim() !== "" && data.password.trim() !== "";
   }
 
-  return { sendLogin, validateForm, formState, countError};
-};
+  const setSubmitButton = (isEnabled: boolean) => {
```

Comment:
Nao entendi mt bem essa nomenclatura, se for usado somente para definir se esta habilitado, algo mais lexico seria legal(talvez isSubmitDisabled). Outro ponto, pelo que vi aqui esta criando uma funcao somente para mudar o nome, uma opcao seria colocar  no objeto de retorno setSubmitButton: setSubmitButtonState que teria o mesmo efeito